scanner.go

```diff
@@ -259,6 +286,19 @@ func (s *scanner) Close() error {
 	return nil
 }
 
+// GetScanMetrics returns the scan metrics for the scanner at the time of call.
+// The scan metrics are non-nil only if the Scan has TrackScanMetrics() enabled
+// The scan metrics for a Scan are only accurate and complete if the scanner has
+// been closed with an io.EOF.
+func (s *scanner) GetScanMetrics() map[string]int64 {
+	if s.scanMetrics == nil {
+		return nil
+	}
+	s.RLock()
+	defer s.RUnlock()
+	return maps.Clone(s.scanMetrics)
```

Comment:
This seems to allow getting the metrics before the scan is complete. Is that useful? Or could we get away with removing the locking around the metrics?